### PR TITLE
Change schemas of /recovery_share GET/POST endpoints

### PR DESCRIPTION
**Pull request by letmaik**
Addresses the second point of https://github.com/microsoft/CCF/issues/1944.

GET /recovery_share now returns `{"encrypted_share": "..base64"}` while POST /recovery_share has as input `{"share": "..base64"}` and returns `{"message": "status message..."}`.

### CODE REVIEW

File: src/node/rpc/call_types.h [hunk 1]
```diff
@@ -155,4 +155,26 @@ namespace ccf
       bool valid = false;
     };
   };
+
+  struct GetRecoveryShare
```
Comment (jumaffre) [inline src/node/rpc/call_types.h:159]: I believe the `submit_recovery_share.sh` script will need updating.

Comment (letmaik) [reply src/node/rpc/call_types.h:159]: Ah, that means semantically what you get and what you send is not the same actually. One is encrypted, one is plaintext. I'll make the fields `"encrypted_share"` and `"share"`, maybe?

Comment (jumaffre) [reply src/node/rpc/call_types.h:159]: Yes, that seems even better!

Comment (letmaik) [reply src/node/rpc/call_types.h:159]: Done.

**Review by achamayou — APPROVED** [review]

**Pull request merged**